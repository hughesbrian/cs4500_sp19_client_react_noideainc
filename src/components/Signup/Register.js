import React from 'react'
import '../../css/Register.scss'
import { Button } from 'react-bootstrap';

const Register = ({first, last, username, email, password, updateField, register}) => {
    return (
        <div className="register-form">
            <h1 className="header">Create Your Account</h1>
            <div className="form">
                <div className="field">
                    <label>First Name</label>
                    <input value={first} name="first" onChange={updateField} className="input-field"></input>
                </div>
                <div className="field">
                    <label>Last Name</label>
                    <input value={last} name="last" onChange={updateField} className="input-field"></input>
                </div>
                <div className="field">
                    <label>Username</label>
                    <input value={username} name="username" onChange={updateField} className="input-field"></input>
                </div>
                <div className="field">
                    <label>Email</label>
                    <input value={email} name="email" onChange={updateField} className="input-field"></input>
                </div>
                <div className="field">
                    <label>Password</label>
                    <input value={password} type="password" name="password" onChange={updateField} className="input-field"></input>
                </div>
                <Button onClick={register} className="register-button">SIGN UP</Button>
            </div>
        </div>
    )
}

export default Register