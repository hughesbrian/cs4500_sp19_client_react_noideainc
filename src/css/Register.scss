.register-form {
    height: 700px;
    width: 65%;
    margin-top: 20px;
    margin-left: 250px;
    border: 1px solid black;
    .header {
        margin-top: 20px;
        text-align: center;
    }
    .form {
        margin-left: 120px;
        margin-top: 20px;
        .field {
            margin-top: 15px;
            margin-left: 100px
        }
        .input-field {
            height: 45px;
            width: 400px;
            display: block;
            border: 1px solid grey
        }
        .register-button {
            height: 40px;
            width: 200px;
            text-align: center;
            float: right;
            margin-top: 35px;
            margin-right: 200px;
            font-size: 20px;
            background-color:#8AD8E3;
            border: none
        }
    }
}