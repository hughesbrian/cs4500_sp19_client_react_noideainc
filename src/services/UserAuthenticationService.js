const baseURL = process.env.NODE_ENV === 'development' ? 'http://localhost:8080' : 'https://cs4500-sp19-noideainc.herokuapp.com'

export default class UserAuthenticationService {
    static instance = null;

    static getInstance() {
        if (UserAuthenticationService.instance === null) {
            UserAuthenticationService.instance = new UserAuthenticationService()
        }
        return this.instance
    }

    login = user => 
        fetch(baseURL + "/api/login", 
            {
                method: 'POST',
                body: JSON.stringify(user),
                headers: {'content-type': 'application/json'},
                credentials: "include"
            }
        ).then(response => {
            return response.json()
        })

    logout = () => 
        fetch(baseURL + "/api/logout", 
            {
                method: 'POST',
                headers: {'content-type': 'application/json'},
                credentials: "include"
            }
        ).then(response => response.json())

    checkLogin = () => 
        fetch(baseURL + "/api/checkLogin", {
            method: 'GET',
            credentials: "include"
        })
        .then(response => {
            return response.json()
        })

}