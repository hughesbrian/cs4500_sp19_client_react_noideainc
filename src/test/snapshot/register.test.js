import React from 'react'
import TestRenderer from 'react-test-renderer'
import users from "../MockData/NewUsers.mock.json"   
import '../mockservice/UserAuthenticationService.mock'
import Register from '../../components/Signup/Register';

test('test sign up', () => {
    const register = () => {
        return users[0]
    }

    const testRenderer = TestRenderer.create(

        <Register 
            first={users[0].firstName} 
            last={users[0].lastName} 
            username={users[0].username}
            email={users[0].email} 
            password={users[0].password}
            register={register}
            />
    )

    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();

    const testInstance = testRenderer.root

    const registerButton = testInstance.findByProps({className: 'register-button'})

    registerButton.props.onClick()

})

test('test render sign up page correctly', () => {

    const testRenderer = TestRenderer.create(
        <Register 
            first={users[0].firstName} 
            last={users[0].lastName} 
            username={users[0].username}
            email={users[0].email} 
            password={users[0].password}
            />
    )

    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();

    const testInstance = testRenderer.root

    const first = testInstance.findByProps({className: "input-field first"})
    const last = testInstance.findByProps({className: "input-field last"})
    const username = testInstance.findByProps({className: "input-field username"})
    const email = testInstance.findByProps({className: "input-field email"})
    const password = testInstance.findByProps({className: "input-field password"})
    const fields = testInstance.findAllByProps({className: "field"})

    expect(first.props.value).toBe('luna')
    expect(last.props.value).toBe('Liu')
    expect(username.props.value).toBe('luna')
    expect(email.props.value).toBe('luna@gmail.com')
    expect(password.props.value).toBe('luna')
    expect(fields.length).toBe(5)

})