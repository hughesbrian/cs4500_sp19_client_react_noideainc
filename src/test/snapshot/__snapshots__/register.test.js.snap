// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render sign up page correctly 1`] = `
<div
  className="register-form"
>
  <h1
    className="header"
  >
    Create Your Account
  </h1>
  <div
    className="form"
  >
    <div
      className="field"
    >
      <label>
        First Name
      </label>
      <input
        className="input-field first"
        name="first"
        value="luna"
      />
    </div>
    <div
      className="field"
    >
      <label>
        Last Name
      </label>
      <input
        className="input-field last"
        name="last"
        value="Liu"
      />
    </div>
    <div
      className="field"
    >
      <label>
        Username
      </label>
      <input
        className="input-field username"
        name="username"
        value="luna"
      />
    </div>
    <div
      className="field"
    >
      <label>
        Email
      </label>
      <input
        className="input-field email"
        name="email"
        value="luna@gmail.com"
      />
    </div>
    <div
      className="field"
    >
      <label>
        Password
      </label>
      <input
        className="input-field password"
        name="password"
        type="password"
        value="luna"
      />
    </div>
    <button
      className="register-button btn btn-primary"
      disabled={false}
      type="button"
    >
      SIGN UP
    </button>
  </div>
</div>
`;

exports[`test sign up 1`] = `
<div
  className="register-form"
>
  <h1
    className="header"
  >
    Create Your Account
  </h1>
  <div
    className="form"
  >
    <div
      className="field"
    >
      <label>
        First Name
      </label>
      <input
        className="input-field first"
        name="first"
        value="luna"
      />
    </div>
    <div
      className="field"
    >
      <label>
        Last Name
      </label>
      <input
        className="input-field last"
        name="last"
        value="Liu"
      />
    </div>
    <div
      className="field"
    >
      <label>
        Username
      </label>
      <input
        className="input-field username"
        name="username"
        value="luna"
      />
    </div>
    <div
      className="field"
    >
      <label>
        Email
      </label>
      <input
        className="input-field email"
        name="email"
        value="luna@gmail.com"
      />
    </div>
    <div
      className="field"
    >
      <label>
        Password
      </label>
      <input
        className="input-field password"
        name="password"
        type="password"
        value="luna"
      />
    </div>
    <button
      className="register-button btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      SIGN UP
    </button>
  </div>
</div>
`;
